src/test/java/org/folio/rest/impl/EholdingsPackagesImplTest.java

```diff
@@ -0,0 +1,98 @@
+
+@RunWith(VertxUnitRunner.class)
+public class EholdingsPackagesImplTest extends WireMockTestBase {
+
+  private static final String PACKAGE_STUB_FILE = "responses/rmapi/packages/get-package-by-id-response.json";
+
+  @Test
+  public void shouldSendDeleteRequestForPackage() throws IOException, URISyntaxException {
+    String providerId = "123355";
+    String packageId = "2884739";
+
+    String wiremockUrl = getWiremockUrl();
+    TestUtil.mockConfiguration(CONFIGURATION_STUB_FILE, wiremockUrl);
+
+    UrlPathPattern packageUrlPattern = new UrlPathPattern(new EqualToPattern("/rm/rmaccounts/" + STUB_CUSTOMER_ID + "/vendors/" + providerId + "/packages/" + packageId), false);
+    EqualToJsonPattern putBodyPattern = new EqualToJsonPattern("{\"isSelected\":false}", true, true);
+
+    WireMock.stubFor(
+      WireMock.get(packageUrlPattern)
+        .willReturn(new ResponseDefinitionBuilder()
+          .withBody(TestUtil.readFile(PACKAGE_STUB_FILE))));
+
+    WireMock.stubFor(
+      WireMock.put(packageUrlPattern)
+        .withRequestBody(putBodyPattern)
+        .willReturn(new ResponseDefinitionBuilder()
+          .withStatus(HttpStatus.SC_NO_CONTENT)));
+
+    RequestSpecification requestSpecification = getRequestSpecification();
+
+    RestAssured.given()
+      .spec(requestSpecification)
+      .when()
+      .delete("eholdings/packages/" + providerId + "-" + packageId)
+      .then()
+      .statusCode(HttpStatus.SC_NO_CONTENT);
+
+    WireMock.verify(1, WireMock.putRequestedFor(packageUrlPattern)
+      .withRequestBody(putBodyPattern));
+  }
+
+  @Test
+  public void shouldReturn400WhenPackageIdIsInvalid() {
+    RequestSpecification requestSpecification = getRequestSpecification();
+    RestAssured.given()
+      .spec(requestSpecification)
+      .when()
+      .delete("eholdings/packages/abc-def")
+      .then()
+      .statusCode(HttpStatus.SC_BAD_REQUEST);
+  }
+
+  @Test
+  public void shouldReturn400WhenPackageIsNotCustom() throws URISyntaxException, IOException {
+    String providerId = "123355";
+    String packageId = "2884739";
+
+    String wiremockUrl = getWiremockUrl();
+    TestUtil.mockConfiguration(CONFIGURATION_STUB_FILE, wiremockUrl);
+
+    ObjectMapper mapper = new ObjectMapper();
+    PackageData packageData = mapper.readValue(TestUtil.getFile(PACKAGE_STUB_FILE), PackageData.class);
+    packageData.setCustom(false);
+
+    WireMock.stubFor(
+      WireMock.get(new UrlPathPattern(new EqualToPattern("/rm/rmaccounts/" + STUB_CUSTOMER_ID + "/vendors/" + providerId + "/packages/" + packageId), false))
+        .willReturn(new ResponseDefinitionBuilder()
+          .withBody(mapper.writeValueAsString(packageData))));
+
+    RequestSpecification requestSpecification = getRequestSpecification();
+
+    RestAssured.given()
+      .spec(requestSpecification)
+      .when()
+      .delete("eholdings/packages/" + providerId + "-" + packageId)
+      .then()
+      .statusCode(HttpStatus.SC_BAD_REQUEST);
+  }
+}
```

Comment:
can we also add one more test to see that a GET of a deleted custom package results in a 404 Not found with error.title = "Package not found"

Reply:
get package by id endpoint is not implemented yet, we will need to add this after we implement it